init.lua

```diff
@@ -146,12 +154,14 @@ minetest.register_on_generated(function(minp, maxp, blockseed)
 		h2 = math.floor(math.min(math.max(math.floor(h2), 0), 100) / 6.6)
 		map = map .. string.char(h1 + (h2 * 16))
 	end
-	local p = string.format("%d", minetest.hash_node_position(v))
-
-	local f = assert(io.open(wp .. "/" .. p, "w"), wp .. "/" .. p)
-	f:write(minetest.compress(map))
-	f:close()
-
+	local p = minetest.hash_node_position(v)
+	if tablesize > 75 then
+		gt = {}
+		collectgarbage()
```

Comment:
This is likely not so bad, but, I think you can avoid calling `collectgarbage()` here entirely. I also think that this method of wiping all cache is relatively inefficient, as you're constantly redoing the caches.

A not-so-complex approach is to keep hashing on `p`, but not embed only the data in there, but also a `timestamp` and delete only old things from the cache.

so `gt[p].timestamp` and `gt[p].value`.

Then you can walk the array, and erase older values than some limit or threshold. 

Perhaps you could even embed a `cache hit` counter in the record and use that to eliminate things that get none or little cache hits, and always keep items with lots of cache hits.

When you do that, you want to make sure you don't remove just 1 item and have to re-prune when 1 item is added, just let the cache grow e.g. to 128, and then prune down to 64.

I would avoid calling `collectgarbage()` if you can at all times and let the lua VM take care of this, if you can.

Reply:
I agree this part can be improved.
The current code downside is that when the cache is discarded there is a chance to see the biome API called to finish the generation.  So about 1 or 2 map every 75 is likely to fall back on the biome API instead. 

These biome API calls could be reduce to nearly zero by pruning half the oldest cache instead. Also removing the collectgarbage() would be nice.